### PR TITLE
[OrderByComparator] LPD-26917 Clean up and Apply Singleton in module portal-workflow, data-engine, dynamic-data-list and dynamic-data-mapping

### PR DESCRIPTION
**Pull request by Valmir-unicap**
- Ticket: https://liferay.atlassian.net/browse/LPD-26917

### CODE REVIEW

**Comment by Valmir-unicap** [comment]
ci:test:sf

**Comment by Valmir-unicap** [comment]
ci:test:relevant

**Comment by Valmir-unicap** [comment]
ci:report:59357232

**Comment by Valmir-unicap** [comment]
ci:test:sf

**Comment by Valmir-unicap** [comment]
ci:test:relevant

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-api/src/main/java/com/liferay/portal/workflow/metrics/util/comparator/WorkflowMetricsSLADefinitionVersionIdComparator.java [hunk 1]
```diff
@@ -14,12 +14,26 @@
 public class WorkflowMetricsSLADefinitionVersionIdComparator
 	extends OrderByComparator<WorkflowMetricsSLADefinitionVersion> {
 
-	public WorkflowMetricsSLADefinitionVersionIdComparator() {
-		this(false);
-	}
+	public static final String ORDER_BY_ASC =
```
Comment (dantewang) [inline modules/dxp/apps/portal-workflow/portal-workflow-metrics-api/src/main/java/com/liferay/portal/workflow/metrics/util/comparator/WorkflowMetricsSLADefinitionVersionIdComparator.java:17]: @Valmir-unicap Why making this change? I think we are going to keep these as is?

**Review by dantewang — CHANGES_REQUESTED** [review]
@Valmir-unicap Please see my comment. Besides that, I don't see other problem.

File: modules/dxp/apps/portal-workflow/portal-workflow-metrics-api/src/main/java/com/liferay/portal/workflow/metrics/util/comparator/WorkflowMetricsSLADefinitionVersionIdComparator.java (hunk 1 above)
Comment (Valmir-unicap) [reply modules/dxp/apps/portal-workflow/portal-workflow-metrics-api/src/main/java/com/liferay/portal/workflow/metrics/util/comparator/WorkflowMetricsSLADefinitionVersionIdComparator.java:17]: Sorry, erro my! I'll fix it now!

**Comment by Valmir-unicap** [comment]
ci:test:sf

**Comment by Valmir-unicap** [comment]
- That failure in `SF`, is not related to my changes.

**Comment by Valmir-unicap** [comment]
ci:test:relevant

**Comment by Valmir-unicap** [comment]
ci:report:61268574

**Comment by Valmir-unicap** [comment]
ci:report:61268574

**Comment by Valmir-unicap** [comment]
ci:report:61268574

**Comment by Valmir-unicap** [comment]
ci:report:61268574

**Comment by dantewang** [comment]
ci:forward:force

**Pull request closed without merging**